edb/edgeql/compiler/expr.py

```diff
@@ -666,3 +692,17 @@ def flatten_set(expr: qlast.Set) -> List[qlast.Expr]:
             elements.append(el)
 
     return elements
+
+
+def collect_binop(expr: qlast.Expr, op: str) -> List[qlast.Expr]:
+    elements = []
+    def _go(expr: qlast.Expr) -> None:
```

Comment:
Perhaps make this into an iterative traversal to further avoid recursion issues?

Reply:
I think I had it in my head that there was some pass before this point that would do a full tree traversal, and so the important bit was just dodging the ~10x stack frame blowup in pgsql. But that's wrong, and this was the first thing hit when I made it much bigger